apps/hashdotai/blog/039_self-building-database.mdx

```diff
@@ -0,0 +1,128 @@
+
+## A graph datastore
+
+To support all of the things we imagine a self-building, globally-connected database needs to be able to do, to be truly useful, we need a new kind of graph database with native support for:
+
+- Multi-tenancy, allowing for the creation of a globally connected knowledge graph, integrated alongside every individual and organization's private data, with the ability to securely and selectively grant access to subgraphs of information when desired.
+- Bitemporal versioning, enabling effective conflict resolution and multi-party access to data.
+- Provenance and confidence metadata, enabling contextual processing of information based on probabilities (taking into account its source, and your own or a consensus view of its trustworthiness, as well as technical factors that might impact individual data's reliability).
+- Strongly typed entities, which support crosswalking via the type system we describe above.
+
+## Agentic AI
+
+To derive structure from the unstructured web we need intelligent agents. These agents need to be capable of running for an extremely long time, processing vast quantities of data. They need to be able to identify real patterns amidst noise, pick out the most important needles within unruly haystacks, and solve problems, perform calculations, and conduct analysis on the fly... introspecting and checking their own work, seeking confirmatory and contradictory sources, and formulating confidence assessments around the "synthesis" they produce.
+
+## A task executor
+
+Finally, to support complex agentic AI, we need a task executor capable of handling extremely long-running jobs. Thankfully, "durable execution" solutions such as _Temporal_ are now widely available, and open-source.
+
+# Solution
+
+**This isn't future technology.**
+
+In fact, [HASH](/) is already being used to solve each of the use cases outlined above.
+
+HASH is a new kind of database for integrating information, which grows, checks, and maintains itself, with a graphical user interface (GUI) that makes its contents directly accessible to all of its end-users. HASH provides all of the capabilities we "imagine" in our introduction.
+
+We founded _HASH, Inc._ as a research lab in 2019. In the years since, we've built the type system, graph, and application (in Rust, for speed, safety, and WASM-compatibility) steadily and quietly, testing it in the real-world throughout.
+
+If you'd like to use HASH yourself, we're now inviting users, and we invite you to [sign up](https://app.hash.ai/signup).
+
+- The [protocol](https://github.com/blockprotocol/blockprotocol) repository contains our **type system**, available under both the MIT and Apache 2.0 Licenses (dually, at your option).
```

Comment:
I'd be tempted to leave this out, and refer to the HASH guide (or failing that, the BP docs) instead as an explanation of the type system... the BP repository is not an accessible guide to the type system (the guide/docs are intended for that, at different levels of technical detail), and we have moved the key packages into the `hash` repository.